File: packages/core/src/types/lane.ts
export interface Lane {
  id: string;
  bayId: string;
  goal: string;
  status: LaneStatus;
  agentId: string | null;
  modelId: string | null;
  fileScope: FileScope;
  checkpoints: string[];
  createdAt: string;
  updatedAt: string;
}

export const LANE_STATUSES = [
  'idle',
  'planning',
  'running',
  'paused',
  'awaiting_approval',
  'reviewing',
  'completed',
  'failed',
  'rolled_back',
] as const;
export type LaneStatus = (typeof LANE_STATUSES)[number];

export function isLaneStatus(value: string): value is LaneStatus {
  return (LANE_STATUSES as readonly string[]).includes(value);
}

export interface FileScope {
  readPatterns: string[];
  writePatterns: string[];
}

File: apps/desktop/src-tauri/src/models/lane.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct Lane {
    pub id: String,
    pub bay_id: String,
    pub goal: String,
    pub status: String,
    pub agent_id: Option<String>,
    pub model_id: Option<String>,
    pub file_scope: Option<String>,
    pub created_at: String,
    pub updated_at: String,
}
